src/main/java/frc/robot/subsystems/RollerSubsystem.java

```diff
@@ -129,34 +125,24 @@ public void setRollPower(double power) {
 
     @Override
     public void periodic() {
-        boolean opening = openTimer.get() > 0 && !openTimer.hasElapsed(OPEN_TIME_SECONDS);
-        boolean closing = !opening || !allowOpen;
-
-        if (closing) {
+        // If we're opening and not allowed to anymore, or if we've finished opening, stop opening and start closing.
+        if ((openTimer.get() > 0 && !allowOpen) || openTimer.hasElapsed(OPEN_TIME_SECONDS)) {
+            openTimer.stop();
+            openTimer.reset();
             closeTimer.start();
         }
 
-        if (closeTimer.get() > 0) {
-            if (!closeTimer.hasElapsed(CLOSE_TIME_SECONDS)) {
-                openMotor.set(-.2);
-            } else {
-                openMotor.set(0);
-            }
-        } else {
-            openMotor.set(0.5);
+        // If we're closing and finished, stop closing.
+        if (closeTimer.hasElapsed(CLOSE_TIME_SECONDS)) {
+            closeTimer.stop();
+            closeTimer.reset();
         }
 
-        // If `OPEN_TIME` hasn't elapsed yet, run the motor.
-        // boolean opening = openTimer.get() > 0 && !openTimer.hasElapsed(OPEN_TIME_SECONDS);
-        // if (opening && allowOpen) {
-        //     openMotor.set(opening && allowOpen ? 0.5 : 0);
-        // }
-
-        // If the cooldown has passed, stop and reset the timer.
-        if (openTimer.hasElapsed(OPEN_TIME_SECONDS + COOLDOWN_SECONDS)) {
-            openTimer.stop();
-            openTimer.reset();
-        }
+        // Otherwise, open if we're opening and close if we're closing.
+        openMotor.set(
+            openTimer.get() > 0 ? 0.5 :
+            closeTimer.get() > 0 ? -0.2 : 0
```

Comment:
i feel that the use of the ternary here combined with the non-intuitive usage of the timer makes this a lil difficult to read

Reply:
What if we created a custom class that wrapped `Timer` to keep track of a `started` boolean and used `hasStarted()` here instead? Would that be easier to read?